parity-util-mem/src/impls.rs

```diff
@@ -73,3 +69,120 @@ impl<T: MallocSizeOf> MallocSizeOf for RwLock<T> {
 		self.read().size_of(ops)
 	}
 }
+
+#[cfg(test)]
+mod tests {
+	use crate::{allocators::new_malloc_size_ops, MallocSizeOf, MallocSizeOfOps};
+	use smallvec::SmallVec;
+	impl_smallvec!(3);
+
+	#[test]
+	fn test_smallvec_stack_allocated_type() {
+		let mut v: SmallVec<[u8; 3]> = SmallVec::new();
+		let mut ops = new_malloc_size_ops();
+		assert_eq!(v.size_of(&mut ops), 0);
+		v.push(1);
+		v.push(2);
+		v.push(3);
+		assert_eq!(v.size_of(&mut ops), 0);
+		assert!(!v.spilled());
+		v.push(4);
+		assert!(v.spilled(), "SmallVec spills when going beyond the capacity of the inner backing array");
+		assert_eq!(v.size_of(&mut ops), 4); // 4 u8s on the heap
+	}
+
+	#[test]
+	fn test_smallvec_boxed_stack_allocated_type() {
+		let mut v: SmallVec<[Box<u8>; 3]> = SmallVec::new();
+		let mut ops = new_malloc_size_ops();
+		assert_eq!(v.size_of(&mut ops), 0);
+		v.push(Box::new(1u8));
+		v.push(Box::new(2u8));
+		v.push(Box::new(3u8));
+		cfg_if::cfg_if! {
+			if #[cfg(any(
+				target_os = "windows",
+				all(target_os = "macos", not(feature = "jemalloc-global")),
+				feature = "estimate-heapsize",
+				feature = "weealloc-global",
+				feature = "dlmalloc-global",
+			))] {
+				assert_eq!(v.size_of(&mut ops), 3); // 3 u8s on the heap, boxes are on the stack
```

Comment:
IMHO it's better to remove conditional compilation and just check `>=` here and below

Reply:
I am not even sure if these tests make any sense at all. It was quite surprising to me to discover how differently the allocators behave for even such simple cases. Out of curiosity I wanted to iterate until I had the right numbers and then sort of just delete them. If you think there’s a value in using >= I’m fine with that too.